// atherdon-newsletter-js-layouts-body/dist/index.es.js
const headlineComponent = (content) => {
    // if (!unsubscribe) new Error('invalid unsubscribe');
    return `content`;
};

const logoBottomComponent = () => {
  return `<table
  border="0"
  cellpadding="0"
  cellspacing="0"
  width="100%"
  class="mcnImageBlock"
  style="
    min-width: 100%;
    border-collapse: collapse;
    mso-table-lspace: 0pt;
    mso-table-rspace: 0pt;
    -ms-text-size-adjust: 100%;
    -webkit-text-size-adjust: 100%;
  "
  >
  <tbody class="mcnImageBlockOuter">
    <tr>
      <td
        valign="top"
        style="
          padding: 9px;
          mso-line-height-rule: exactly;
          -ms-text-size-adjust: 100%;
          -webkit-text-size-adjust: 100%;
        "
        class="mcnImageBlockInner"
      >
        <table
          align="left"
          width="100%"
          border="0"
          cellpadding="0"
          cellspacing="0"
          class="mcnImageContentContainer"
          style="
            min-width: 100%;
            border-collapse: collapse;
            mso-table-lspace: 0pt;
            mso-table-rspace: 0pt;
            -ms-text-size-adjust: 100%;
            -webkit-text-size-adjust: 100%;
          "
        >
          <tbody>
            <tr>
              <td
                class="mcnImageContent"
                valign="top"
                style="
                  padding-right: 9px;
                  padding-left: 9px;
                  padding-top: 0;
                  padding-bottom: 0;
                  text-align: center;
                  mso-line-height-rule: exactly;
                  -ms-text-size-adjust: 100%;
                  -webkit-text-size-adjust: 100%;
                "
              >
                <a
                  href="https://hackernoon.com/"
                  title=""
                  class=""
                  target="_blank"
                  style="
                    mso-line-height-rule: exactly;
                    -ms-text-size-adjust: 100%;
                    -webkit-text-size-adjust: 100%;
                  "
                >
                  <img
                    align="center"
                    alt=""
                    src="https://creative-images-upld.s3.amazonaws.com/creative/newsletters/logos/brand/hackernoon.png"
                    width="564"
                    style="
                      max-width: 15426px;
                      padding-bottom: 0;
                      display: inline !important;
                      vertical-align: bottom;
                      border: 0;
                      height: auto;
                      outline: none;
                      text-decoration: none;
                      -ms-interpolation-mode: bicubic;
                    "
                    class="mcnImage"
                  />
                </a>
              </td>
            </tr>
          </tbody>
        </table>
      </td>
    </tr>
  </tbody>
  </table>`;
};

const logoTopComponent = () => {
  return `<table
  border="0"
  cellpadding="0"
  cellspacing="0"
  width="100%"
  class="mcnImageBlock"
  style="
    min-width: 100%;
    border-collapse: collapse;
    mso-table-lspace: 0pt;
    mso-table-rspace: 0pt;
    -ms-text-size-adjust: 100%;
    -webkit-text-size-adjust: 100%;
  "
  >
  <tbody class="mcnImageBlockOuter">
    <tr>
      <td
        valign="top"
        style="
          padding: 0px;
          mso-line-height-rule: exactly;
          -ms-text-size-adjust: 100%;
          -webkit-text-size-adjust: 100%;
        "
        class="mcnImageBlockInner"
      >
        <table
          align="left"
          width="100%"
          border="0"
          cellpadding="0"
          cellspacing="0"
          class="mcnImageContentContainer"
          style="
            min-width: 100%;
            border-collapse: collapse;
            mso-table-lspace: 0pt;
            mso-table-rspace: 0pt;
            -ms-text-size-adjust: 100%;
            -webkit-text-size-adjust: 100%;
          "
        >
          <tbody>
            <tr>
              <td
                class="mcnImageContent"
                valign="top"
                style="
                  padding-right: 0px;
                  padding-left: 0px;
                  padding-top: 0;
                  padding-bottom: 0;
                  text-align: center;
                  mso-line-height-rule: exactly;
                  -ms-text-size-adjust: 100%;
                  -webkit-text-size-adjust: 100%;
                "
              >
                <a
                  href="https://www.hackernoon.com"
                  title=""
                  class=""
                  target="_blank"
                  style="
                    mso-line-height-rule: exactly;
                    -ms-text-size-adjust: 100%;
                    -webkit-text-size-adjust: 100%;
                  "
                >
                  <img
                    align="center"
                    alt=""
                    src="https://creative-images-upld.s3.amazonaws.com/creative/newsletters/logos/brand/hackernoon.png"
                    width="600"
                    style="
                      max-width: 1200px;
                      padding-bottom: 0;
                      display: inline !important;
                      vertical-align: bottom;
                      border: 0;
                      height: auto;
                      outline: none;
                      text-decoration: none;
                      -ms-interpolation-mode: bicubic;
                    "
                    class="mcnImage"
                  />
                </a>
              </td>
            </tr>
          </tbody>
        </table>
      </td>
    </tr>
  </tbody>
  </table>`;
};

const previewTextComponent = (content) => {
  
  return `<span
  class="mcnPreviewText"
  style="
    display: none;
    font-size: 0px;
    line-height: 0px;
    max-height: 0px;
    max-width: 0px;
    opacity: 0;
    overflow: hidden;
    visibility: hidden;
    mso-hide: all;
  "
  >${content}</span>`;
};

const sectionComponent = (content) => {
  return `
  <table
                      border="0"
                      cellpadding="0"
                      cellspacing="0"
                      width="100%"
                      class="mcnTextBlock"
                      style="
                        min-width: 100%;
                        border-collapse: collapse;
                        mso-table-lspace: 0pt;
                        mso-table-rspace: 0pt;
                        -ms-text-size-adjust: 100%;
                        -webkit-text-size-adjust: 100%;
                      "
                    >
                      <tbody class="mcnTextBlockOuter">
                        <tr>
                          <td
                            valign="top"
                            class="mcnTextBlockInner"
                            style="
                              padding-top: 9px;
                              mso-line-height-rule: exactly;
                              -ms-text-size-adjust: 100%;
                              -webkit-text-size-adjust: 100%;
                            "
                          >
                            <!--[if mso]>
          <table align="left" border="0" cellspacing="0" cellpadding="0" width="100%" style="width:100%;">
          <tr>
          <![endif]-->
  
                            <!--[if mso]>
          <td valign="top" width="600" style="width:600px;">
          <![endif]-->
                            <table
                              align="left"
                              border="0"
                              cellpadding="0"
                              cellspacing="0"
                              style="
                                max-width: 100%;
                                min-width: 100%;
                                border-collapse: collapse;
                                mso-table-lspace: 0pt;
                                mso-table-rspace: 0pt;
                                -ms-text-size-adjust: 100%;
                                -webkit-text-size-adjust: 100%;
                              "
                              width="100%"
                              class="mcnTextContentContainer"
                            >
                              <tbody>
                                <tr>
                                  <td
                                    valign="top"
                                    class="mcnTextContent"
                                    style="
                                      padding: 0px 18px 9px;
                                      line-height: 150%;
                                      mso-line-height-rule: exactly;
                                      -ms-text-size-adjust: 100%;
                                      -webkit-text-size-adjust: 100%;
                                      word-break: break-word;
                                      color: #111111;
                                      font-family: 'Source Sans Pro',
                                        'Helvetica Neue', Helvetica, Arial,
                                        sans-serif;
                                      font-size: 18px;
                                      text-align: left;
                                    "
                                  >
                                    ${content}
                                  </td>
                                </tr>
                              </tbody>
                            </table>
                            <!--[if mso]>
          </td>
          <![endif]-->
  
                            <!--[if mso]>
          </tr>
          </table>
          <![endif]-->
                          </td>
                        </tr>
                      </tbody>
                    </table>`;
};

const sponsorComponent = (href, src, content) => {
  return 
  
};

const ctaComponent = () => {
    return `<div dir="ltr" style="text-align: justify;"><strong style="font-weight: bolder;">Ready to Claim Your Internet Name on Hacker Noon?</strong><ol><li dir="ltr" style="mso-line-height-rule: exactly;-ms-text-size-adjust: 100%;-webkit-text-size-adjust: 100%;"><p dir="ltr" role="presentation" style="line-height: 125%;margin: 10px 0;padding: 0;mso-line-height-rule: exactly;-ms-text-size-adjust: 100%;-webkit-text-size-adjust: 100%;color: #111111;font-family: 'Source Sans Pro', 'Helvetica Neue', Helvetica, Arial, sans-serif;font-size: 18px;text-align: left;"><a href="https://hackernoon.com/signup?ref=noonifications.tech" target="_blank" style="mso-line-height-rule: exactly;-ms-text-size-adjust: 100%;-webkit-text-size-adjust: 100%;color: #111111;font-weight: bold;text-decoration: underline;">Create a free account</a>.</p></li><li dir="ltr" style="mso-line-height-rule: exactly;-ms-text-size-adjust: 100%;-webkit-text-size-adjust: 100%;"><p dir="ltr" role="presentation" style="line-height: 125%;margin: 10px 0;padding: 0;mso-line-height-rule: exactly;-ms-text-size-adjust: 100%;-webkit-text-size-adjust: 100%;color: #111111;font-family: 'Source Sans Pro', 'Helvetica Neue', Helvetica, Arial, sans-serif;font-size: 18px;text-align: left;"><a href="https://app.hackernoon.com/subscriptions?ref=noonifications.tech" target="_blank" style="mso-line-height-rule: exactly;-ms-text-size-adjust: 100%;-webkit-text-size-adjust: 100%;color: #111111;font-weight: bold;text-decoration: underline;">Tell us what you want to read about</a>.</p></li><li dir="ltr" style="mso-line-height-rule: exactly;-ms-text-size-adjust: 100%;-webkit-text-size-adjust: 100%;"><p dir="ltr" role="presentation" style="line-height: 125%;margin: 10px 0;padding: 0;mso-line-height-rule: exactly;-ms-text-size-adjust: 100%;-webkit-text-size-adjust: 100%;color: #111111;font-family: 'Source Sans Pro', 'Helvetica Neue', Helvetica, Arial, sans-serif;font-size: 18px;text-align: left;"><a href="https://app.hackernoon.com/new?ref=noonifications.tech" target="_blank" style="mso-line-height-rule: exactly;-ms-text-size-adjust: 100%;-webkit-text-size-adjust: 100%;color: #111111;font-weight: bold;text-decoration: underline;">Submit your own tech stories</a>. </p></li><li dir="ltr" style="mso-line-height-rule: exactly;-ms-text-size-adjust: 100%;-webkit-text-size-adjust: 100%;"><p dir="ltr" role="presentation" style="line-height: 125%;margin: 10px 0;padding: 0;mso-line-height-rule: exactly;-ms-text-size-adjust: 100%;-webkit-text-size-adjust: 100%;color: #111111;font-family: 'Source Sans Pro', 'Helvetica Neue', Helvetica, Arial, sans-serif;font-size: 18px;text-align: left;"><a href="https://sponsor.hackernoon.com/brand-as-author?ref=noonifications.tech" target="_blank" style="mso-line-height-rule: exactly;-ms-text-size-adjust: 100%;-webkit-text-size-adjust: 100%;color: #111111;font-weight: bold;text-decoration: underline;">Gain Access to Brand-As-Author</a>™ Program. </p></li><li dir="ltr" style="mso-line-height-rule: exactly;-ms-text-size-adjust: 100%;-webkit-text-size-adjust: 100%;"><p dir="ltr" role="presentation" style="line-height: 125%;margin: 10px 0;padding: 0;mso-line-height-rule: exactly;-ms-text-size-adjust: 100%;-webkit-text-size-adjust: 100%;color: #111111;font-family: 'Source Sans Pro', 'Helvetica Neue', Helvetica, Arial, sans-serif;font-size: 18px;text-align: left;">Get Your Tech Stories Featured on the <a href="https://hackernoon.com/?ref=noonifications.tech" target="_blank" style="mso-line-height-rule: exactly;-ms-text-size-adjust: 100%;-webkit-text-size-adjust: 100%;color: #111111;font-weight: bold;text-decoration: underline;">Homepage of Hacker Noon</a>.</p></li></ol></div>`;};

const body = {
  headlineComponent,
  logoBottomComponent,
  logoTopComponent,
  previewTextComponent,
  sectionComponent,
  sponsorComponent,
  ctaComponent
};

export { body as default };
